docs/SIL.rst

```diff
@@ -5522,6 +5522,14 @@ The second operand may have either object or address type.  In the
 latter case, the dependency is on the current value stored in the
 address.
 
+The optional ``nonescaping`` attribute indicates that no value derived
+from ``value`` escapes the lifetime of ``base``. Derived values
+include copies and values transitively forwarded from the original or
+any copy. If ``base`` is identical to ``value`` this simply means that
+copies do not outlive the original ``value``.  Furthermore, no value
+derived from ``value`` may have a bitwise escape (conversion to
+UnsafePointer) or pointer escapes (unknown use).
```

Comment:
I'm not sure I understood the meaning of the nonescaping flag.
Even without this flag, the value must not escape the base. It follows from "Operations that would destroy the second value must not be moved before any instructions which depend on the result of this instruction".
From the PR description I assume the flag means that the non-escaping property of the value must be statically visible in SIL. Is this right? If so, you should describe it here.

Reply:
Literally nothing in this documentation block is true with regular mark_dependence

```
The optional ``nonescaping`` attribute indicates that no value derived
from ``value`` escapes the lifetime of ``base``. Derived values
include copies and values transitively forwarded from the original or
any copy. If ``base`` is identical to ``value`` this simply means that
copies do not outlive the original ``value``.  Furthermore, no value
derived from ``value`` may have a bitwise escape (conversion to
```

Reply:
I rewrote the entire doc section to make this clear:

```
  sil-instruction :: 'mark_dependence' '[nonescaping]'? sil-operand 'on' sil-operand

  %2 = mark_dependence %value : $*T on %base : $Builtin.NativeObject

Indicates that the validity of ``%value`` depends on the value of
``%base``. Operations that would destroy ``%base`` must not be moved
before any instructions which depend on the result of this
instruction, exactly as if the address had been directly derived from
that operand (e.g. using ``ref_element_addr``).

The result is the forwarded value of ``%value``. ``%value`` may be an
address, but it could be an address in a non-obvious form, such as a
Builtin.RawPointer or a struct containing the same.

``%base`` may have either object or address type. In the latter case,
the dependency is on the current value stored in the address.

The optional ``nonescaping`` attribute indicates that no value derived
from ``%value`` escapes the lifetime of ``%base``. As with escaping
``mark_dependence``, all values transitively forwarded from ``%value``
must be destroyed within the lifetime of ``%base``. Unlike escaping
``mark_dependence``, this must be statically verifiable. Additionally,
unlike escaping ``mark_dependence``, derived values include copies of
``%value`` and values transitively forwarded from those copies. If
``%base`` is identical to ``%value`` this simply means that copies of
``%value`` do not outlive the original OSSA lifetime of
``%value``. Furthermore, unlike escaping ``mark_dependence``, no value
derived from ``%value`` may have a bitwise escape (conversion to
UnsafePointer) or pointer escape (unknown use).
```